core/src/player.rs

```diff
@@ -2268,8 +2268,8 @@ impl PlayerBuilder {
 
     /// Sets the storage backend of the player.
     #[inline]
-    pub fn with_storage(mut self, storage: impl 'static + StorageBackend) -> Self {
-        self.storage = Some(Box::new(storage));
+    pub fn with_storage(mut self, storage: Box<dyn StorageBackend>) -> Self {
```

Comment:
Since `Player` and especially `PlayerBuilder` are kinda-sorta de-facto "public API", this broke the Android app:
https://github.com/ruffle-rs/ruffle-android/actions/runs/8599069910/job/23561194287#step:6:42
:/

I really wish this would have followed the `with_renderer` / `with_boxed_renderer` right above it.
Anyway, no big deal, I can fix it, this breakage just feels unnecessary and inconsistent.